task1.py

```diff
@@ -0,0 +1,65 @@
+class Task:  # класс задания
+    def __init__(self, name: str, reward: int):
+        self.name = name        # имя задания
+        self.reward = reward        # награда за задание
+    @property
+    def name(self):
+        return self._name
+
+    @name.setter
+    def name(self, new_name: str):
+        if not isinstance(new_name, str):
+            raise TypeError('Имя должно быть типа str')
+        self._name = new_name
+
+    # конвертация награды в золото
+    def convert_to_gold(self):
+        return self.reward * 100
+
+    def __str__(self):
+        return f'Задание "{self._name}". Награда - {self.reward}'
+
+    def __repr__(self):
+        return f'{self.__class__.__name__}(name={self._name!r}, reward={self.reward!r})'
+
+
+class Dungeon(Task):  # класс подземелья
+    def __init__(self, name: str, difficult: str, reward: int):
+        super().__init__(name, reward)
+        super.__str__(name)
+        super().convert_to_gold()
```

Comment:
```suggestion
        self.add_reward()
```
В данном случае было бы корректно указать что будет изменение награды